src/groups/group_action.jl

```diff
@@ -0,0 +1,90 @@
+"""
+    AbstractActionOnManifold
+
+An abstract group action on a manifold.
+"""
+abstract type AbstractActionOnManifold end
+
+"""
+    base_group(A::AbstractActionOnManifold)
+
+The group that acts in action `A`.
+"""
+function base_group(A::AbstractActionOnManifold)
+    error("Function base_group is not yet defined for type $(typeof(A)).")
+end
+
+"""
+    action_on(A::AbstractActionOnManifold)
+
+The manifold the action `A` acts upon.
+"""
+function action_on(A::AbstractActionOnManifold)
```

Comment:
This name is confusing, because it doesn't return the action, it takes the action. I don't have a better idea. The technical name for the manifold upon which a group acts is I believe a G-manifold.

Reply:
We can rename it to `g_manifold` then?

Reply:
So we could rename this to `g_manifold`?

Reply:
I think so. It's an annoying name, but it's consistent with the usual terminology.